src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/features/form/validation/singleField/singleFieldValidationSagas.ts

```diff
@@ -17,28 +18,31 @@ export function* runSingleFieldValidationSaga(): SagaIterator {
     state.instanceData.instance,
   );
   const url = getDataValidationUrl(state.instanceData.instance.id, currentTaskDataId);
+  const currentSingleFieldValidation = state.formValidations.currentSingleFieldValidation;
 
-  if (state.formValidations.currentSingleFieldValidation) {
+  if (currentSingleFieldValidation) {
     const options: AxiosRequestConfig = {
       headers: {
-        ValidationTriggerField: state.formValidations.currentSingleFieldValidation,
+        ValidationTriggerField: currentSingleFieldValidation,
       },
     };
 
     try {
+      // Reset current single field validation for next potential validation
+      yield put(setCurrentSingleFieldValidation({ dataModelBinding: null }));
+
       const serverValidation: any = yield call(get, url, options);
```

Comment:
Are we getting the single-field validations every time we are blurring a field here? And what about generic validations, are we getting all validations, such as valid input etc, from backend? What happens if we do not receive the serverValidation rules? Does the field pass as valid?

Reply:
We get single-field validations on blur if the field has been set up to trigger them: https://altinn.github.io/docs/altinn-studio/app-creation/logic/validation/#enkeltfeltvalidering.
This is used for custom validations that go beyond simple validation against data model.
In addition, client-side validations are performed separately for checking things like format/length/type etc. against the data model.
If there are no errors client-side, and we do not receive validations from backend, then the field would pass as valid. However, a complete server-side validation of all fields is performed before submit of form, so if the data really is invalid it would be caught at this point.